sumcheck/src/prover/plain.rs

```diff
@@ -0,0 +1,182 @@
+/// the sum-check prover for the following two statements
+/// $$
+/// p_{\nu - \kappa}\left(v_{\kappa+1}, \cdots, v_{\nu}\right) = \sum_{w_i}
+///     EQ\left(\left(v_{\kappa+1}, \cdots, v_{\nu}\right), \left(w_{\kappa+1}, \cdots,
+///                              w_{\nu}\right)\right) \cdot
+///      \left( p_{\nu-\kappa+1}\left(1, w_{\kappa+1}, \cdots, w_{\nu}\right)  \cdot
+///              q_{\nu-\kappa+1}\left(0, w_{\kappa+1}, \cdots, w_{\nu}\right) +
+///     p_{\nu-\kappa+1}\left(0, w_{\kappa+1}, \cdots, w_{\nu}\right)  \cdot
+///     q_{\nu-\kappa+1}\left(1, w_{\kappa+1}, \cdots, w_{\nu}\right)\right)
+/// $$
+///
+/// and
+///
+/// $$
+/// q_{\nu -k}\left(v_{\kappa+1}, \cdots, v_{\nu}\right) = \sum_{w_i}EQ\left(\left(v_{\kappa+1},
+///  \cdots, v_{\nu}\right), \left(w_{\kappa+1}, \cdots, w_{\nu }\right)\right) \cdot
+/// \left( q_{\nu-\kappa+1}\left(1, w_{\kappa+1}, \cdots, w_{\nu}\right)  \cdot
+///  q_{\nu-\kappa+1}\left(0, w_{\kappa+1}, \cdots, w_{\nu}\right)\right)
+/// $$
+///
+/// for $k = 1, \cdots, \nu - 1$  
+///
+/// Instead of executing two runs of the sum-check protocol, a batching randomness `r_batch` is
+/// sent by the verifier at the outset in order to batch the two statments.
+///
+/// [1]: https://eprint.iacr.org/2023/1284
+#[allow(clippy::too_many_arguments)]
+pub fn sumcheck_prove_plain<E: FieldElement, H: ElementHasher<BaseField = E::BaseField>>(
+    claim: E,
+    r_batch: E,
+    p0: &mut MultiLinearPoly<E>,
+    p1: &mut MultiLinearPoly<E>,
+    q0: &mut MultiLinearPoly<E>,
+    q1: &mut MultiLinearPoly<E>,
+    eq: &mut MultiLinearPoly<E>,
+    transcript: &mut impl RandomCoin<Hasher = H, BaseField = E::BaseField>,
+) -> Result<SumCheckProof<E>, SumCheckProverError> {
+    let mut round_proofs = vec![];
+
+    let mut claim = claim;
+    let mut challenges = vec![];
+
+    for _ in 0..p0.num_variables() {
+        let len = p0.num_evaluations() / 2;
+
+        #[cfg(not(feature = "concurrent"))]
+        let (eval_point_1, eval_point_2, eval_point_3) =
+            (0..len).fold((E::ZERO, E::ZERO, E::ZERO), |(a, b, c), i| {
+                let eval_point_1 = comb_func(
+                    p0[2 * i + 1],
+                    p1[2 * i + 1],
+                    q0[2 * i + 1],
+                    q1[2 * i + 1],
+                    eq[2 * i + 1],
+                    r_batch,
+                );
+
+                let p0_delta = p0[2 * i + 1] - p0[2 * i];
+                let p1_delta = p1[2 * i + 1] - p1[2 * i];
+                let q0_delta = q0[2 * i + 1] - q0[2 * i];
+                let q1_delta = q1[2 * i + 1] - q1[2 * i];
+                let eq_delta = eq[2 * i + 1] - eq[2 * i];
+
+                let mut p0_evx = p0[2 * i + 1] + p0_delta;
+                let mut p1_evx = p1[2 * i + 1] + p1_delta;
+                let mut q0_evx = q0[2 * i + 1] + q0_delta;
+                let mut q1_evx = q1[2 * i + 1] + q1_delta;
+                let mut eq_evx = eq[2 * i + 1] + eq_delta;
```

Comment:
Would rename those to e.g. `p0_eval_at_x` or just `p0_at_x`. I never remember what `evx` stands for